src/machine/stack.rs

```diff
@@ -242,7 +242,8 @@ impl Stack {
     #[inline(always)]
     pub(crate) fn index_and_frame_mut(&mut self, e: usize) -> &mut AndFrame {
         unsafe {
-            let ptr = self.buf.base as usize + e;
+            // This is doing alignment wrong
```

Comment:
In particular, Miri catches ptr here with alignment 4 instead of the expected 8. Creating unaligned references is UB, but I think this works because unaligned memory access is allowed in x86_64, it's just slower.